docs/_articles/en/v3.x/salesforce-graph-engine/rules.md

```diff
@@ -243,46 +239,132 @@ These common scenarios trigger a violation from MultipleMassSchemaLookupRule.
 
-*Explanation*
+Explanation:
 
 `Schema.getGlobalDescribe` or `Schema.describeSObjects` is executed multiple times in a single path. Reduce the execution of the method to one time, then rescan your code.
 
-### UnusedMethodRule
+### UnimplementedTypeRule <a name='UnimplementedTypeRule'>#</a>
 
-UnusedMethodRule is a path-based analysis rule that detects many methods contained in your code that aren’t invoked from any entry points that Graph Engine recognizes. UnusedMethodRule detects methods contained in your code that aren’t invoked. It detects:
+UnimplementedTypeRule detects abstract classes and interfaces that are non-global and missing implementations or extensions.
 
-- static methods
-- instance methods
+#### Definition
 
-To invoke the path-based rules run: `scanner:run --engine sfge --projectdir MyDirectory`. 
+UnimplementedTypeRule is a traditional static analysis rule where a violation occurs at a point in the code where the interface or abstract class is declared. It doesn’t use sources or sinks.
 
-**Example**: 
-```sfdx scanner:run --engine sfge --projectdir /project/dir --target /project/dir/target1```
+#### Interpreting UnimplementedTypeRule Results
 
-UnusedMethodRule recognizes these entry points.
+Match any violation message that you receive with this case to understand more about the violation.
+
+*Common Case*
+
+> Extend, implement, or delete %s %s
+
+Explanation:
+
+Because this abstract class or interface has no implementations or extensions, it can’t be instantiated. It’s unnecessary and can be deleted.
+
+#### UnimplementedTypeRule Limitations
+
+Because UnimplementedType rule excludes `global` scoped classes from consideration, these classes are prevented from being thrown as false positives and aren’t false negatives.
+
+## Pilot Rules
+
+### AvoidDatabaseOperationInLoop <a name='AvoidDatabaseOperationInLoop'>#</a>
+
+AvoidDatabaseOperationInLoop detects database operations that occur inside loops and which cause performance degradation. Database operations within loops cause performance degradations and exceed Salesforce Governor Limits. 
+
+To remediate database operation violations, move these operations outside of loops. Follow these best practices.
+
+* Instead of querying objects within a loop, use a SOQL For Loop to iterate over a query’s results..
+
+For example, instead of:
+
+```
+for (String name : names) {
+   Account[] accs = [SELECT Name FROM Account WHERE Name =: name];
+```
+
+   Use:
+```
+for (Account[] accs : [SELECT Name FROM Account WHERE Name IN :names]) {
+```
+
+* Instead of putting database operations within a loop, use the loop to iteratively construct lists of objects, and then perform the database operation on the full list after the loop.
+
+For example, instead of:
+
+```
+for (String n : names) {
+   insert new Account(Name = n);
+```
+
+   Use:
+```
+Account[] accs = new List<Account>();
+for (String n : names) {
+   accs.add(new Account(Name = n));
+}
+insert accs;
+```
```

Comment:
In this block, if you indent the two lines starting with `For example` and the four lines that have the triple tick (```), then the entire block will render indented, which keeps it in line with the bullet points.

Reply:
done